Restor/Utilities/HashUtils.m

```diff
@@ -16,24 +16,89 @@
 
 #import <CommonCrypto/CommonDigest.h>
 
+// Returns the hexadecimal string representation for an array of bytes.
+// @param bytes is a byte array.
+// @param len is the number of bytes in the array.
+NSString *hexStringFromBytes(unsigned char *bytes, int len) {
+  const char *digits = "0123456789abcdef";
```

Comment:
Why not:

```objc
NSMutableString *s = [[NSMutableString alloc] initWithCapacity:len];
for (int i = 0; i < len; ++i) {
  [s appendFormat:@"%02X", bytes[i]];
}
return s
```

Which is a bit easier to read. It's probably a tiny bit slower but in the context of hashing a 4GB file a couple of milliseconds is negligible.

Reply:
I agree that it's not really noticeable in this context since we're talking about microseconds, but that version is about 23 times slower than the other for a 256-bit digest and 34 times slower for a 512-bit digest.  This seems like too high a price for brevity to me, but I can change it if you feel strongly about it.

Reply:
I don't feel particularly strongly